rotkehlchen/chain/ethereum/uniswap/uniswap.py

```diff
@@ -110,6 +124,80 @@ def __init__(
                 f'Probably will get fixed with time. If not report it to Rotkis support channel ',
             )
 
+    @staticmethod
+    def _calculate_events_balances(
+            address: ChecksumEthAddress,
+            events: List[LiquidityPoolEvent],
+            balances: List[LiquidityPool],
+    ) -> List[LiquidityPoolEventsBalance]:
+        """Given an address, its LP events and the current LPs participating in
+        (`balances`), process each event (grouped by pool) aggregating the
+        token0, token1 and USD amounts for calculating the profit/loss in the
+        pool. Finally return a list of <LiquidityPoolEventsBalance>, where each
+        contains the profit/loss and events per pool.
+
+        If `balances` is empty that means either the address does not have
+        balances in the protocol or the endpoint has been called with a
+        specific time range.
+        """
+        events_balances: List[LiquidityPoolEventsBalance] = []
+        pool_balance: Dict[ChecksumEthAddress, LiquidityPool] = (
+            {pool.address: pool for pool in balances}
+        )
+        pool_aggregated_amount: Dict[ChecksumEthAddress, AggregatedAmount] = {}
+        # Populate `pool_aggregated_amount` dict, being the keys the pools'
+        # addresses and the values the aggregated amounts from their events
+        for event in events:
+            pool = event.pool_address
+
+            if pool not in pool_aggregated_amount:
+                pool_aggregated_amount[pool] = AggregatedAmount()
+
+            pool_aggregated_amount[pool].events.append(event)
+
+            if event.event_type == EventType.MINT:
+                pool_aggregated_amount[pool].profit_loss0 += FVal(event.amount0)
```

Comment:
A question I have when looking at this is why not have all amounts and prices already made to FVals when creating the events? Would it be much cleaner like that?

----

After checking all the PR's code again it seems you already have them as Fval's due to `AssetAmount`.

So are you doingit again due to  mypy complaining? If yes then please add a type: ignore and make an issue for this.

There is something wrong with these types like `AssetAmount`, `Price` e.t.c. which is a kind of a wrapper type. We use it to have a bit of a more strict form of typing, but we want it to be seen as a type of FVal and in some places it's not.

Perhaps the right way would be to make "fake types" by superclassing FVal. Not sure ... we have to research a bit.That's why I say add ignores here and make an issue.

If we fix it, mypy will complain about the unused ignores and we can remove them.

Reply:
Nah, you were totally right, types not needed!